File: Backend/src/main/java/com/app/zakat/model/Transaction.java
package com.app.zakat.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "transaction")
@Data
public class Transaction {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int transactionId;
	
	private String value;
	
	private LocalDateTime date;
	
	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;
	
	@ManyToOne
	@JoinColumn(name = "currency_id")
	private Currency currency;
	
	@ManyToOne
	@JoinColumn(name = "revenue_id")
	private Revenue revenue;
}

File: Backend/Domain/Entities/Transaction.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities;

public partial class Transaction
{
    public int TransactionId { get; set; }

    public DateTime? Date { get; set; }

    public string? Value { get; set; }

    public int? CurrencyId { get; set; }

    public int? RevenueId { get; set; }

    public int? UserId { get; set; }

    public virtual Currency? Currency { get; set; }

    public virtual Revenue? Revenue { get; set; }

    public virtual User? User { get; set; }
}
